### PR TITLE
Add support for filtering multiple paths

### PR DESCRIPTION
**Pull request by raniejade**

### CODE REVIEW

File: spek-runtime/jvm/src/main/kotlin/org/spekframework/spek2/runtime/SpekJvmRuntime.kt [hunk 1]
```diff
@@ -38,15 +38,13 @@ actual class SpekRuntime: AbstractRuntime() {
             .filter { !it.isAbstract }
             .map { klass ->
                 klass to PathBuilder.from(klass).build()
+            }.map { (klass, path) ->
```
Comment (arturdryomov) [inline spek-runtime/jvm/src/main/kotlin/org/spekframework/spek2/runtime/SpekJvmRuntime.kt]: Let’s follow the same style everywhere here, i. e.

```kotlin
.map { 
}
.filter { }
.map { }
```

File: spek-runtime/jvm/src/main/kotlin/org/spekframework/spek2/runtime/SpekJvmRuntime.kt [hunk 2]
```diff
@@ -38,15 +38,13 @@ actual class SpekRuntime: AbstractRuntime() {
...
+                val matched = discoveryRequest.paths.firstOrNull { it.isRelated(path) }
+                matched to resolveSpec(instanceFactoryFor(klass).create(klass), path)
```
Comment (arturdryomov) [inline spek-runtime/jvm/src/main/kotlin/org/spekframework/spek2/runtime/SpekJvmRuntime.kt]: I think that there is some redundancy here. Even if path was not matched you are calling `resolveSpec` for it.

File: spek-runtime/jvm/src/main/kotlin/org/spekframework/spek2/runtime/SpekJvmRuntime.kt (hunk 1 above)
Comment (raniejade) [reply spek-runtime/jvm/src/main/kotlin/org/spekframework/spek2/runtime/SpekJvmRuntime.kt]: 👍

File: spek-runtime/jvm/src/main/kotlin/org/spekframework/spek2/runtime/SpekJvmRuntime.kt (hunk 2 above)
Comment (raniejade) [reply spek-runtime/jvm/src/main/kotlin/org/spekframework/spek2/runtime/SpekJvmRuntime.kt]: 👍

**Comment by raniejade** [comment]
@ming13 changes in!

**Review by arturdryomov — APPROVED** [review]

**Pull request merged**